### ISSUE TITLE
`cached_property` is saved even if it has not been called yet

### ISSUE THREAD

**Issue by pheuer** [varchasgopalaswamy/prepper#4]
My understanding was that, if a `cached_property` has not been called yet, it should not be called during export. My reason for this is that all the information may not be present (e.g. if the cached property is part of a later analysis step), so it's possible that calling the cached property could raise exceptions. 

Currently it looks like cached properties are called and run when they are exported, even if they haven't been accessed yet? See test

```
def test_export_import_cached_property(tmp_path):
    """
    Test exporting and importing a cached property 
    """
    
    @saveable_class("0.0.1")
    class ClassWithCachedProperty(ExportableClassMixin):
        """
        This class has a cached property
        """
        _exportable_attributes = ['expensive_fcn']
        
        @cached_property
        def expensive_fcn(self):
            return 'This string took 1000 hours to calculate'
        
        
    path = os.path.join(tmp_path, 'tmp.hdf5')
    
    obj1 = ClassWithCachedProperty()
    obj1.to_hdf5(path)
    
    # So far we haven't actually called the cached_property, so it should
    # NOT be saved?
    
    with h5py.File(path, 'r') as f:
        print(list(f['expensive_fcn'].attrs.keys()))
        print(f['expensive_fcn'][...])
```

**Comment by varchasgopalaswamy** [comment]
Good catch. Not sure why this regression occurred; I'll take a look and fix it. It's definitely not intended behavior!

**Comment by pheuer** [comment]
Has this issue been patched?

**Comment by varchasgopalaswamy** [comment]
Nope. I am doing another pass through prepper tomorrow and should have this and other lingering issues fixed.

**Comment by varchasgopalaswamy** [comment]
@pheuer this is fixed in 1ba0d9a
